src/fairdrive-protocol.ts

```diff
@@ -39,46 +46,78 @@ export class FairdriveProtocol {
       this.users[username] = address
     } else if (mnemonic) {
       try {
-        this.users[username] = Wallet.fromMnemonic(mnemonic).address
+        const wallet = Wallet.fromMnemonic(mnemonic)
+        this.users[username] = wallet.address
+        this.setActiveAccount(wallet)
       } catch (e) {
         throw new Error('Incorrect mnemonic')
       }
     }
   }
 
   /**
-   * Logs in with the FDP credentails and gives back ethers wallet
+   * Logs in with the FDP credentials and gives back ethers wallet
    *
    * @param username FDP username
    * @param password password of the wallet
    * @returns BIP-039 + BIG-044 Wallet
    */
   async userLogin(username: string, password: string): Promise<Wallet> {
-    if (!this.users[username]) {
-      throw new Error('User is not imported')
-    }
+    validateUsername(username)
+    validatePassword(password)
+
+    const address = this.users[username]
 
-    if (!password) {
-      throw new Error('Empty password')
+    if (!address) {
+      throw new Error('User is not imported')
     }
 
-    const address = this.users[username]
-    const encryptedMnemonic = await getEncryptedMnemonic(this.bee, username, address)
+    const encryptedMnemonic = await getEncryptedMnemonic(this.accountData.bee, username, address)
     const decrypted = decrypt(password, encryptedMnemonic.text())
 
     try {
-      return Wallet.fromMnemonic(decrypted)
+      const wallet = Wallet.fromMnemonic(decrypted)
+      this.setActiveAccount(wallet)
+
+      return wallet
     } catch (e) {
       throw new Error('Incorrect password')
     }
   }
 
   async userSignup(username: string, password: string, mnemonic = ''): Promise<UserAccountWithReference> {
-    // todo check input
     // todo check is already exists / imported
-    const userInfo = await createUser(this.bee, username, password, mnemonic)
+    validateUsername(username)
+    validatePassword(password)
+
+    const userInfo = await createUser(this.accountData, username, password, mnemonic)
     this.users[username] = userInfo.wallet.address
+    this.setActiveAccount(userInfo.wallet)
 
     return userInfo
   }
+
+  async podLs(): Promise<Pod[]> {
+    validateActiveAccount(this.accountData)
+
+    const result = await getFeedData(
+      this.accountData.bee,
+      FairdriveProtocol.POD_TOPIC,
+      this.accountData.wallet!.address,
```

Comment:
that can cause error, can't it? since the wallet is optional maybe it is undefined.

Reply:
The previous line checks that there is an active account and a wallet is present.